src/middleware/messageStore.coffee

```diff
@@ -105,31 +88,72 @@ exports.storeResponse = (ctx, done) ->
     # reset request body - primary route
     res.body = ''
 
-    # reset request body - routes
-    if ctx.routes
-      for route in ctx.routes
-        route.response.body = ''
-
-
-  # assign new transactions status to ctx object
-  ctx.transactionStatus = status
-
-  update = { response: res, status: status }
-  update.routes = ctx.routes if ctx.routes?
+  update = { response: res }
 
   if ctx.mediatorResponse
     update.orchestrations = ctx.mediatorResponse.orchestrations if ctx.mediatorResponse.orchestrations
     update.properties = ctx.mediatorResponse.properties if ctx.mediatorResponse.properties
 
-  transactions.Transaction.findOneAndUpdate { _id: ctx.transactionId }, update, (err, tx) ->
+  transactions.Transaction.findOneAndUpdate { _id: ctx.transactionId }, update , { runValidators: true }, (err, tx) ->
+    logger.info "stored primary response for #{tx._id}"
     if err
-      logger.error 'Could not save response metadata for transaction: ' + ctx.transactionId + '. ' + err
+      logger.info 'Could not save response metadata for transaction: ' + ctx.transactionId + '. ' + err
       return done err
     if tx is undefined or tx is null
       logger.error 'Could not find transaction: ' + ctx.transactionId
       return done err
     return done()
 
+exports.storeNonPrimaryResponse = (ctx, done) ->
+  # check if channel response body is false and remove
+  if ctx.authorisedChannel.responseBody == false
+    response.response.body = ''
+
+  if ctx.transactionId?
+    transactions.Transaction.findByIdAndUpdate ctx.transactionId, { routes: ctx.routes }, (err,tx) ->
```

Comment:
This seems to save all of the routes each time its called. I think we would rather want to save only one route at a time instead of all of them.